engine/src/main/java/org/terasology/engine/world/block/BlockPart.java

```diff
@@ -32,38 +29,36 @@
         this.side = side;
     }
 
-    static {
-        SIDE_MAP = new EnumMap<>(Side.class);
-        SIDE_MAP.put(Side.BOTTOM, BOTTOM);
-        SIDE_MAP.put(Side.RIGHT, RIGHT);
-        SIDE_MAP.put(Side.LEFT, LEFT);
-        SIDE_MAP.put(Side.BACK, BACK);
-        SIDE_MAP.put(Side.FRONT, FRONT);
-        SIDE_MAP.put(Side.TOP, TOP);
-
-        SIDES = new BlockPart[]{TOP, LEFT, RIGHT, FRONT, BACK, BOTTOM};
-
-        HORIZONTAL_SIDES = ImmutableList.of(LEFT, RIGHT, FRONT, BACK);
-    }
-
     public static BlockPart fromSide(Side side) {
-        return SIDE_MAP.get(side);
-    }
-
-    /**
-     * @return The block parts corresponding to sides (so not the center)
-     */
-    public static BlockPart[] sideValues() {
-        return SIDES;
+        switch (side) {
+            case BOTTOM:
+                return BOTTOM;
+            case RIGHT:
+                return RIGHT;
+            case LEFT:
+                return LEFT;
+            case BACK:
+                return BACK;
+            case FRONT:
+                return FRONT;
+            case TOP:
+                return TOP;
+            default:
+                break;
+        }
+        return null;
```

Comment:
Do you think returning null is a good idea here? We should at least document this behavior. I'd prefer throwing an exception in the `default` case here, as it should never occur.

```suggestion
            default:
                throw new IllegalArgumentException();
        }
```

**Edit:** Wait, there's the `CENTER` block part which has no corresponding side... :thinking: but I still think that the `default` case cannot appear.